### ISSUE TITLE
indent rule fails to catch unindented object literal

### ISSUE THREAD

**Issue by rhys-vdw** [palantir/tslint#3782]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const x = {
y: 5,
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
   // ...
  "rules": {
     // ...
    "indent": [true, "spaces", 2],
     // ...
  }
}

```

#### Actual behavior

No error

#### Expected behavior

Error on unindented key

**Comment by yarinr** [comment]
I believe the indent rule is only suppose to detect usage of the wrong kind of indentation (as in a tabs when it was configured to only allow spaces).

It does not enforce "correct" indentation of constructs. Your complaint is not unique to object literals:
```typescript
if (true) {
console.log("Hello!");
}
```
Will not raise an error - and the same goes for any other unindented construct.

**Comment by rhys-vdw** [comment]
> I believe the indent rule is only suppose to detect usage of the wrong kind of indentation (as in a tabs when it was configured to only allow spaces).

Ah, then what is [`no-irregular-whitespace`](https://palantir.github.io/tslint/rules/no-irregular-whitespace/)? The docs don't really explain what it does.

**Comment by suchanlee** [comment]
`no-irregular-whitespace` is inspired by https://eslint.org/docs/rules/no-irregular-whitespace. We can work to improve the docs. For code formatting, try using tools like https://github.com/prettier/prettier

**Comment by KaelWD** [comment]
See #2814
